### PR TITLE
UIDEXP-180: Modify transformation form to ensure proper data entry for MARC Bib mappings

### PR DESCRIPTION
**Pull request by yelyzavetakhokhlova**
## Purpose
In the scope of [UIDEXP-180](https://issues.folio.org/browse/UIDEXP-180).
To prevent incorrect data entry and make the transformation form in the mapping profile consistent with Quick MARC and data import apps, the transformation textbox should was divided to separate MARC field indicators and subfields.
The transformation text box is split into 4 parts:

- MARC tag text box
- First indicator text box
- Second indicator text box
- Subfield text box

## Jest coverage
image
image

### CODE REVIEW

File: test/jest/helpers/getTransformationFieldGroups.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {
+  getByTestId,
+  screen,
+} from '@testing-library/react';
+
+export const getTransformationFieldGroups = () => {
```
Comment (ViktorSoroka07) [inline test/jest/helpers/getTransformationFieldGroups.js:6]: That is nice.

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationFieldGroup/TransformationFieldGroup.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      </div>
+    </div>
+  );
+};
+
+TransformationFieldGroup.propTypes = {
+  record: PropTypes.shape({
+    displayName: PropTypes.string.isRequired,
+    order: PropTypes.number.isRequired,
+  }),
```
Comment (ViktorSoroka07) [inline src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationFieldGroup/TransformationFieldGroup.js:65]: Should it be required?

File: src/settings/MappingProfiles/MappingProfilesFormContainer/processRawTransformations.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { omit } from 'lodash';
+
+export const parseRawTransformation = transformation => {
+  // TODO: remove this check once transformations validation is in place.
```
Comment (ViktorSoroka07) [inline src/settings/MappingProfiles/MappingProfilesFormContainer/processRawTransformations.js:4]: Please add ticket number.

**Review by ViktorSoroka07 — APPROVED** [review]

File: test/jest/helpers/getTransformationFieldGroups.js (hunk 1 above)
Comment (yelyzavetakhokhlova) [reply test/jest/helpers/getTransformationFieldGroups.js:6]: ikr 🙃

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationFieldGroup/TransformationFieldGroup.js (hunk 2 above)
Comment (yelyzavetakhokhlova) [reply src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationFieldGroup/TransformationFieldGroup.js:65]: Yep, will do.

**Review by ViktorSoroka07 — APPROVED** [review]
Great job.

File: test/jest/helpers/getTransformationFieldGroups.js (hunk 1 above)
Comment (yelyzavetakhokhlova) [reply test/jest/helpers/getTransformationFieldGroups.js:6]: Thank you!

**Pull request merged**